cmd/cli/trafficpolicy_check.go

```diff
@@ -0,0 +1,205 @@
+	if err != nil {
+		return errors.Errorf("Error listing SMI TrafficTarget policies: %s", err)
+	}
+
+	var foundTrafficTarget bool
+	for _, trafficTarget := range trafficTargets.Items {
+		spec := trafficTarget.Spec
+		// Map traffic targets to the given pods
+		if spec.Destination.Name == dstPod.Spec.ServiceAccountName && spec.Destination.Namespace == dstPod.Namespace {
+			// The TrafficTarget destination is associated to 'dstPod'
+
+			// Check if 'srcPod` is an allowed source to this destination
+			for _, source := range spec.Sources {
+				if source.Name == srcPod.Spec.ServiceAccountName && source.Namespace == srcPod.Namespace {
+					fmt.Fprintf(cmd.out, "[+] Pod '%s/%s' is allowed to communicate to pod '%s/%s' via the SMI TrafficTarget policy %q:\n",
+						srcPod.Namespace, srcPod.Name, dstPod.Namespace, dstPod.Name, trafficTarget.Name)
+					foundTrafficTarget = true
+
+					target := trafficTarget // avoids gosec G601: Implicit memory aliasing in for loop
+					trafficTargetPolicy, err := yaml.Marshal(&target)
+					if err != nil {
+						return errors.Errorf("Failed to marshal TrafficTarget %s: %s", trafficTarget.Name, err)
+					}
+					fmt.Fprintf(cmd.out, "---\n%s\n---\n", string(trafficTargetPolicy))
+				}
+			}
+		}
+	}
+
+	if !foundTrafficTarget {
+		fmt.Fprintf(cmd.out, "[+] Pod '%s/%s' is not allowed to communicate to pod '%s/%s', missing SMI TrafficTarget policy\n",
+			srcPod.Namespace, srcPod.Name, dstPod.Namespace, dstPod.Name)
+	}
+
+	return nil
+}
+
+func (cmd *trafficPolicyCheckCmd) getMeshedPod(namespace, podName string) (*corev1.Pod, error) {
+	// Validate the pods
+	pod, err := cmd.clientSet.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Errorf("Could not find pod %s in namespace %s", podName, namespace)
+	}
+	if !isMeshedPod(*pod) {
+		return nil, errors.Errorf("Pod %s in namespace %s is not a part of a mesh", podName, namespace)
+	}
+	return pod, nil
+}
+
+func (cmd *trafficPolicyCheckCmd) isPermissiveModeEnabled() (bool, error) {
+	osmNamespace := settings.Namespace()
+	configMap, err := cmd.clientSet.CoreV1().ConfigMaps(osmNamespace).Get(context.TODO(), osmConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return false, errors.Errorf("Error checking if permissive mode is enabled: %s", err)
+	}
+	strval, ok := configMap.Data[permissiveTrafficPolicyModeKey]
+	if !ok {
+		return false, errors.Errorf("Missing key %q in %s/%s ConfigMap", permissiveTrafficPolicyModeKey, configMap.Namespace, configMap.Name)
+	}
+	configMapBoolValue, err := strconv.ParseBool(strval)
+	if err != nil {
+		return false, errors.Errorf("Invalid value %q for key %q in %s/%s ConfigMap, must be boolean", strval, permissiveTrafficPolicyModeKey, configMap.Namespace, configMap.Name)
+	}
+	return configMapBoolValue, nil
+}
+
+func unmarshalNamespacedPod(namespacedPod string) (string, string, error) {
```

Comment:
It is a little tough to understand what is being returned here with the 1 strings.

To make this easier to comprehend I would create explicit variable names that suggest what this is. For instance:
```go
namespace := chunk[0]
podName := chunk[1]
return namespace, podName
```

And of course we could use named return values - tells you what you get from this function at first glance.

Reply:
named return is perfect for this tiny function. Will update.